scripts/scil_tractogram_point_math.py

```diff
@@ -0,0 +1,182 @@
+    p = argparse.ArgumentParser(
+        formatter_class=argparse.RawTextHelpFormatter,
+        description=__doc__)
+
+    p.add_argument('operation', metavar='OPERATION',
+                   choices=['mean', 'sum', 'min',
+                            'max', 'correlation'],
+                   help='The type of operation to be performed on the '
+                        'streamlines. Must\nbe one of the following: '
+                        '%(choices)s.')
+    p.add_argument('in_tractogram', metavar='INPUT_FILE',
+                   help='Input tractogram containing streamlines and '
+                        'metadata.')
+    p.add_argument('--in_dpp_name',  nargs='+', action='append',
+                   required=True,
+                   help='Name of the data_per_point for operation to be '
+                        'performed on.')
+    p.add_argument('--out_name', nargs='+', action='append',
+                   required=True,
+                   help='Name of the resulting data_per_point or '
+                   'data_per_streamline to be saved in the output '
+                   'tractogram. (Default: %(default)s)')
+    p.add_argument('out_tractogram', metavar='OUTPUT_FILE',
+                   help='The file where the remaining streamlines '
+                        'are saved.')
+
+    p.add_argument('--endpoints_only', action='store_true', default=False,
+                   help='If set, will only perform operation on endpoints \n'
+                   'If not set, will perform operation on all streamline \n'
+                   'points.')
+
+    add_reference_arg(p)
+    add_verbose_arg(p)
+    add_overwrite_arg(p)
+    add_bbox_arg(p)
+
+    return p
+
+
+def main():
+    parser = _build_arg_parser()
+    args = parser.parse_args()
+
+    if args.verbose:
+        logging.getLogger().setLevel(logging.INFO)
+
+    assert_inputs_exist(parser, args.in_tractogram)
+    assert_outputs_exist(parser, args, args.out_tractogram)
+
+    # Load the input files.
+    logging.info("Loading file {}".format(args.in_tractogram))
+    sft = load_tractogram_with_reference(parser, args, args.in_tractogram)
+
+    if len(sft.streamlines) == 0:
+        logging.info("Input tractogram contains no streamlines. Exiting.")
+        return
+
+    if len(args.in_dpp_name[0]) != len(args.out_name[0]):
+        parser.error('The number of in_dpp_names and out_names must be '
+                     'the same.')
+
+    data_per_point = {}
+    data_per_streamline = {}
+    for in_dpp_name, out_name in zip(args.in_dpp_name[0],
+                                     args.out_name[0]):
+        # Check to see if the data per point exists.
+        if in_dpp_name not in sft.data_per_point:
+            logging.info('Data per point {} not found in input tractogram.'
+                         .format(in_dpp_name))
+            return
+
+        # check size of first streamline data_per_point
+        if sft.data_per_point[in_dpp_name][0].shape[1] == 1:
```

Comment:
What happens if it is a vector of shape (N, 1)? You will think it is singular.

Also, this will crash if the data is not 2D. Ex, if it is of shape (1,), you don't have a shape[1].